### PR TITLE
make fs operations async

### PR DESCRIPTION
**Pull request by Yanpas**
Making filesystem operations in node in a sync manner is a bad practise. For vscode entire exntesion may get unresponsive for several hundres ms. 

Now most of the operations are async, try it out! :slightly_smiling_face:

### CODE REVIEW

File: src/client/modules/q-conn-manager.ts [hunk 1]
```diff
@@ -162,24 +162,24 @@ export class QConnManager {
...
                     if (res) {
                         if (typeof res.r === 'string' && res.r.startsWith('ERROR')) {
                             const msg: string[] = res.r.split('\n');
                             QueryConsole.current?.appendError(msg, duration, uniqLabel, query);
-                            HistoryTreeItem.appendHistory({ uniqLabel: uniqLabel, time: timestamp, duration: duration, query: query, errorMsg: res.r });
+                            await HistoryTreeItem.appendHistory({ uniqLabel: uniqLabel, time: timestamp, duration: duration, query: query, errorMsg: res.r });
                         } else if (QConnManager.consoleMode) {
                             QueryConsole.current?.append(res.r, duration, uniqLabel, query);
-                            HistoryTreeItem.appendHistory({ uniqLabel: uniqLabel, time: timestamp, duration: duration, query: query, errorMsg: '' });
+                            await HistoryTreeItem.appendHistory({ uniqLabel: uniqLabel, time: timestamp, duration: duration, query: query, errorMsg: '' });
```
Comment (Yanpas) [inline src/client/modules/q-conn-manager.ts:182]: Maybe this `await` is unnecessary, just preserved behaviour

**Comment by jshinonome** [comment]
Thanks for the PR, please revisit it and remove unnecessary changes.

**Comment by Yanpas** [comment]
Removed unnecessary and tested the extension locally with q binary

**Comment by jshinonome** [comment]
It takes 1.6 seconds to analyze about 350 files. I don't think there is an issue with the file operation, so I am not going to approve this.

**Comment by Yanpas** [comment]
Took a deeper look, the server runs in a separate process, so it doesn't interfere with extensionhost process. Yet I still think it will benefit from being concurrent, for example after parsing 1 of 350 files it will be able to respond to clients. Since nodejs has only one thread of execution, the language server won't do anything else except for parsing q files, no matter how many are there.
https://rclayton.silvrback.com/avoid-synchronous-functions-in-node-js

Anyway I reverted all server changes and left only client ones.

**Comment by jshinonome** [comment]
Thanks for PR. I closed this PR is not because you are wrong. I don't have a proper way to test vscode-q for now, so I have to check all changes. One more thing is unless it becomes an issue, no need to change it yet.

**Comment by Yanpas** [comment]
"Don't fix if it's working", makes sense, agree

**Pull request closed without merging**